File: CS/PreviewSortSample/MainViewModel.cs
using DevExpress.Mvvm.DataAnnotations;
using DevExpress.Xpf.Printing;
using DevExpress.XtraReports.UI;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Windows;
using System.Windows.Input;

namespace PreviewSortSample {
    [POCOViewModel]
    public class MainViewModel {
        public virtual OrderDetailsReport Report { get; protected set; }
        protected MainViewModel() {
            Report = new OrderDetailsReport();
            Report.SetOrderID(11077);
        }

        public void OnPreviewMouseClick(DocumentPreviewMouseEventArgs args) {
            var sortField = args.ElementTag as string;
            if(!string.IsNullOrEmpty(sortField)) {
                Report.Sort(sortField);
            }
        }

        public void OnPreviewMouseMove(DocumentPreviewMouseEventArgs args) {
            var sortField = args.ElementTag as string;
            if(!string.IsNullOrEmpty(sortField)) {
                Mouse.SetCursor(Cursors.Hand);
            }
        }
    }
}

File: VB/MainViewModel.vb
Imports DevExpress.Mvvm.DataAnnotations
Imports DevExpress.Xpf.Printing
Imports DevExpress.XtraReports.UI
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports System.Windows
Imports System.Windows.Input

Namespace PreviewSortSample
    <POCOViewModel> _
    Public Class MainViewModel
        Private privateReport As OrderDetailsReport
        Public Overridable Property Report() As OrderDetailsReport
            Get
                Return privateReport
            End Get
            Protected Set(ByVal value As OrderDetailsReport)
                privateReport = value
            End Set
        End Property
        Protected Sub New()
            Report = New OrderDetailsReport()
            Report.SetOrderID(11077)
        End Sub

        Public Sub OnPreviewMouseClick(ByVal args As DocumentPreviewMouseEventArgs)
            Dim sortField = TryCast(args.ElementTag, String)
            If Not String.IsNullOrEmpty(sortField) Then
                Report.Sort(sortField)
            End If
        End Sub

        Public Sub OnPreviewMouseMove(ByVal args As DocumentPreviewMouseEventArgs)
            Dim sortField = TryCast(args.ElementTag, String)
            If Not String.IsNullOrEmpty(sortField) Then
                Mouse.SetCursor(Cursors.Hand)
            End If
        End Sub
    End Class
End Namespace
